internal/envoy/v3/auth.go

```diff
@@ -60,6 +60,21 @@ func UpstreamTLSContext(peerValidationContext *dag.PeerValidationContext, sni st
 	return context
 }
 
+// UpstreamTLSContext creates an envoy_v3_tls.UpstreamTlsContext which specifies
+// minimum TLS protocol version and set of cipher suites to be used for upstream connections.
+// By default UpstreamTLSContext returns a HTTP/1.1 TLS enabled context. A list of
+// additional ALPN protocols can be provided.
+func UpstreamTLSContextWithMinProtocolVersion(
+	peerValidationContext *dag.PeerValidationContext, sni string, clientSecret *dag.Secret, minProtocolVersion string, cipherSuites []string,
+	alpnProtocols ...string) *envoy_v3_tls.UpstreamTlsContext {
+	context := UpstreamTLSContext(peerValidationContext, sni, clientSecret, alpnProtocols...)
+	context.CommonTlsContext.TlsParams = &envoy_v3_tls.TlsParameters{
+		TlsMinimumProtocolVersion: ParseTLSVersion(minProtocolVersion),
+		CipherSuites:              cipherSuites,
+	}
+	return context
+}
```

Comment:
Instead of defining a new function to handle the additional parameters, I might define an `UpstreamTLSContextOptions` struct that holds all of the possible parameters, e.g.:

```go
type UpstreamTLSContextOptions struct {
	PeerValidationContext *dag.PeerValidationContext
	SNI                   string
	ClientSecret          *dag.Secret
	ALPNProtocols         []string
	MinimumTLSVersion     string
	CipherSuites          []string
}
```

And then refactor the existing function to look like:

```go
func UpstreamTLSContext(opts UpstreamTLSContextOptions) *envoy_v3_tls.UpstreamTlsContext {
	...
}
```

Where you can check for the existence of each setting in the Options struct. That way it's just one function that can handle the presence/absence of any of the options.

It's not used in *too* many places so the refactor shouldn't be too bad.

This can be non-blocking, it's easy enough to refactor later too.